src/app/page.tsx

```diff
@@ -7,34 +7,54 @@ import EmblaCarousel from './components/EmblaCarousel';
 import { OPTIONS, SLIDES } from './index';
 import {auth} from "@/firebase/firebaseConfig";
 import { useAuthState } from 'react-firebase-hooks/auth';
-import {getAllPortfolios, userHasPortfolios} from "@/functions/databaseAccess";
+import {getAllPortfolios, getPortfolios, userHasPortfolios} from "@/functions/databaseAccess";
 import Pagination from '@/app/components/Pagination';
+import {where} from "@firebase/firestore";
 
 
+function startsWithIgnoreCase(str: string, prefix: string): boolean {
+    return str.toLowerCase().startsWith(prefix.toLowerCase());
+}
+
 export default function Home() {
   const [user, loading] = useAuthState(auth);
+  //Search form values
+  const [name, setName] = useState('');
+  
   const router = useRouter();
 
   let [portfolios, setPortfolios] = useState<{photo: string, fullname: string, role: string}[] | null>(null);
+  const [portfoliosToShow, setPortfoliosToShow] = useState<{photo: string, fullname: string, role: string}[] | null>(null);
   const [currentPage, setCurrentPage] = useState(1);
   const itemsPerPage = 6;
 
   let indexOfLastItem;
   let indexOfFirstItem;
   let currentItems: {photo: string, fullname: string, role: string}[] = [];
 
-  if (portfolios) {
+  if (portfoliosToShow) {
     indexOfLastItem = currentPage * itemsPerPage;
     indexOfFirstItem = indexOfLastItem - itemsPerPage;
-    currentItems = portfolios.slice(indexOfFirstItem, indexOfLastItem);
+    currentItems = portfoliosToShow.slice(indexOfFirstItem, indexOfLastItem);
   }
 
   useEffect(() => {
-    getAllPortfolios().then((data) => {
+    getPortfolios([]).then((data) => {
```

Comment:
You've refactored getAllPortfolios that does getPortfolios([]) in it, why you use the getPortfolios([]) call here?

Reply:
If you are interested, I was trying to create a mechanism for query customization so that we can define predicates for portfolios filtering on our page, and then pass the created predicate object to our function that transforms it into some query and retrieves data. However, it was proven that in our case filtering data on the client side is more effective than filtering it by queries, so I abandoned the query customization approach.